File: go/design-patterns/abstractfactory/naive/screen_naive.go
func renderLoginScreen(os string) []string {
    // ❌ one dev writes the button branch…
    var button Button
    if os == "mac" { button = MacButton{} } else { button = WindowsButton{} }

    // ❌ …another writes the checkbox branch, defaulting to Mac:
    var checkbox Checkbox
    if os == "windows" { checkbox = WindowsCheckbox{} } else { checkbox = MacCheckbox{} }

    // 💥 On Windows: Windows button next to a Mac checkbox.
    return []string{button.Render(), checkbox.Render()}
}

File: cpp/design-patterns/abstractfactory/naive/screen_naive.cpp
std::vector<std::string> renderLoginScreen(const std::string& os) {
    // ❌ one dev writes the button branch…
    auto button = os == "mac"
        ? std::unique_ptr<Button>(new MacButton())
        : std::unique_ptr<Button>(new WindowsButton());

    // ❌ …another writes the checkbox branch, defaulting to Mac:
    auto checkbox = os == "windows"
        ? std::unique_ptr<Checkbox>(new WindowsCheckbox())
        : std::unique_ptr<Checkbox>(new MacCheckbox());

    // 💥 On Windows: Windows button next to a Mac checkbox.
    return { button->render(), checkbox->render() };
}